pkg/lwip/contrib/sock/lwip_sock.c

```diff
@@ -490,11 +490,9 @@ int lwip_sock_get_addr(struct netconn *conn, struct _sock_tl_ep *ep, u8_t local)
         ) {
         return res;
     }
-#if LWIP_IPV6 && LWIP_IPV4
-    ep->family = (addr.type == IPADDR_TYPE_V6) ? AF_INET6 : AF_INET;
-#elif LWIP_IPV6
+#if LWIP_IPV6
     ep->family = (conn->type & NETCONN_TYPE_IPV6) ? AF_INET6 : AF_INET;
-#elif LWIP_IPV4
+#else
     ep->family = AF_INET;
 #endif
```

Comment:
How about doing this?

```C
    if (IS_ACTIVE(LWIP_IPV6) && IS_ACTIVE(LWIP_IPV4)) {
        ep->family = (conn->type & NETCONN_TYPE_IPV6) ? AF_INET6 : AF_INET;
    }
    else if (IS_ACTIVE(LWIP_IPV6) && (conn->type & NETCONN_TYPE_IPV6)) {
        ep->family = AF_INET6;
    }
    else if (IS_ACTIVE(LWIP_IPV4)) {
        ep->family = AF_INET;
    }
    else {
        ep->family = AF_UNSPEC;
    }
```

Reply:
If `LWIP_IPV4` is set and only `LWIP_IPV4` is set, the family will then be set to IPv4 without check. This optimization is not present for IPv6: It will check `conn->type`, even if only IPv6 is enabled.

Reply:
I just realized that `NETCONN_TYPE_IPV6` isn't provided without `LWIP_IPV6` set :-/

Reply:
But `NETCONNTYPE_ISIPV6(x)` is provided in any case :-)

Reply:
Then let's use that. From several versions updates with member renames or (re-)movals, I learned its always better to use lwIP's helper macros than not ;-).

Reply:
How about [this](https://github.com/RIOT-OS/RIOT/pull/14665/commits/cca685a78f9e6a09382aebfc4374100208fcb34e)?